src/arviz_plots/visuals/__init__.py

```diff
@@ -59,9 +60,13 @@ def trace_rug(da, target, backend, mask, xname=None, y=None, **kwargs):
             xvalues = da[xname]
         if y is None:
             y = da.min().item()
+    if flatten is not False:
+        xvalues = xvalues.values.flatten()  # flatten xvalues
     if len(xvalues.shape) != 1:
         raise ValueError(f"Expected unidimensional data but got {xvalues.sizes}")
```

Comment:
can you try what happens when xvalues is always flattened and no error is risen?